app/react/translations/edit/editors/HTMLEditor/RichEditor.js

```diff
@@ -0,0 +1,124 @@
+/* @flow */
+import React from 'react';
+import { Editor, EditorState, RichUtils, ContentState, convertFromHTML } from 'draft-js';
+import type { EditorState as EditorStateType } from 'draft-js';
+import { stateToHTML } from 'draft-js-export-html';
+
+import BlockStyleControls from './components/BlockStyleControls';
+import InlineStyleControls from './components/InlineStyleControls';
+import ToggleHTMLButton from './components/ToggleHTMLButton';
+
+type PropTypes = {
+  onChange: Function,
+  toggleRawEdit: Function,
+  value: string,
+  editAsRaw: boolean,
+  fieldInfo: Object
+}
+
+type StateTypes = {
+  editorState: EditorStateType
+}
+
+const getBlockStyle = block =>
+  (block.getType() === 'blockquote') && 'RichEditor-blockquote';
+
+export default class RichEditor extends React.Component {
+  constructor(props: PropTypes) {
+    super(props);
+    const markup = convertFromHTML(props.value);
+    const editorState: EditorStateType = ContentState.createFromBlockArray(
+      markup.contentBlocks,
+      markup.entityMap
+    );
+    this.state = { editorState: EditorState.createWithContent(editorState) };
+    this.focus = () => this.editor.focus();
+    this.handleKeyCommand = (command: string) => this._handleKeyCommand(command);
+    this.onTab = (e: Event) => this._onTab(e);
+    this.onChange = (newEditorState: EditorStateType) => this._handleChange(newEditorState);
+    this.toggleBlockType = (type: string) => this._toggleBlockType(type);
+    this.toggleInlineStyle = (style: string) => this._toggleInlineStyle(style);
+  }
+
+  state: StateTypes
+  onTab: Function
+  onChange: Function
+  handleKeyCommand: Function
+  focus: Function
+  toggleInlineStyle: Function
+  toggleBlockType: Function
+  editor: HTMLElement
+  props: PropTypes
+
+  _handleKeyCommand(command: string) {
+    const { editorState } = this.state;
+    const newState = RichUtils.handleKeyCommand(editorState, command);
+    if (newState) {
+      this.onChange(newState);
+      return true;
+    }
+    return false;
+  }
+  _onTab(e: Event) {
+    const maxDepth = 4;
```

Comment:
That is Draft.js way of describing "how nested" elements are. In this context, onTab will respect only 4 levels of nestage (if that's even a word).